client/rest-high-level/src/test/java/org/elasticsearch/client/asyncsearch/AsyncSearchIT.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.asyncsearch;
+
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.client.ESRestHighLevelClientTestCase;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class AsyncSearchIT extends ESRestHighLevelClientTestCase {
+
+    public void testSubmitAsyncSearchRequest() throws IOException {
+        String index = "test-index";
+        createIndex(index, Settings.EMPTY);
+
+        SearchRequest searchRequest = new SearchRequest(index);
+        // TODO either check that in client request validation already or set it automatically?
+        searchRequest.setCcsMinimizeRoundtrips(false);
```

Comment:
We should probably not force users to remember to set this to `false`, although `true` seems to be the default. I wonder if we shold either 
* set the value to `false` automatically for every SearchRequest once it is set for an AsyncSearchRequest
* ignore this parameter as request parameter when sending the request
* ignore this paramter on the receiving Rest action regardless of how it is set in the client (maybe warn)

Reply:
Can we copy the original `SubmitAsyncSearchRequest` that creates the `SearchRequest` internally and sets the default appropriately. We can then add a validation client side if needed or let the server-side validation throw error if users override the ccs option ?

Reply:
Thanks, I didn't see we implicitely set the defaulfs in the server side `org.elasticsearch.xpack.core.search.action.SubmitAsyncSearchRequest` we should do that on the client side as well then. Also see we're overwriting these properties:
```
request.setPreFilterShardSize(1);
request.setBatchedReduceSize(5);
request.requestCache(true);
```
I assume we also want to do that then on the client side as well?

Reply:
yes

Reply:
This can be removed now that we set these values automatically in the ctr ?